cmd/zdb/zdb.c

```diff
@@ -4187,6 +4196,264 @@ same_metaslab(spa_t *spa, uint64_t vdev, uint64_t off1, uint64_t off2)
 	return ((off1 >> ms_shift) == (off2 >> ms_shift));
 }
 
+/*
+ * Prints the elements in each row properly for -P output.
+ */
+static inline void
+prtrow_nicenum(uint64_t number, int width)
+{
+	char numbuf[32];
+
+	zdb_nicenum(number, numbuf, sizeof (numbuf));
+	if (dump_opt['P'])
+		(void) printf("\t%s", numbuf);
+	else
+		printf("%*s", width, numbuf);
+}
```

Comment:
nit: this small helper function doesn't buy us much readability.  I'd be inclined to fold in to `dump_size_histograms()`.